packages/compartment-mapper/src/types.js

```diff
@@ -328,3 +328,35 @@ export {};
  * @property {string} name
  * @property {Array<string>} path
  */
+
+/**
+ * @typedef {object} FullAttenuationDefinition
+ * @property {string} name
+ * @property {Array<any>} params
+ */
+
+/**
+ * @typedef {Array<any>} ImplicitAttenuationDefinition
+ */
+
+/**
+ * @typedef {FullAttenuationDefinition | ImplicitAttenuationDefinition} AttenuationDefinition
+ */
+
+/**
+ * @typedef {object} UnifiedAttenuationDefinition
+ * @property {string} displayName
+ * @property {string | null} specifier
+ * @property {Array<any> | undefined} params
```

Comment:
If this is an optional property (not just one that is explicitly `undefined`), you can express that more idiomatically with:

```suggestion
 * @property {Array<any>} [params]
```